### PR TITLE
[Test] Try to enable testharness.js based tests

### PR DESCRIPTION
**Pull request by zqzhang**
The idea is to
- create W3C testharness.js (see files in `resources\`) based test cases for these RealSense APIs which don't require real 3D camera support, for example, tests can be as simple as `realsense.js` checking existence/type of an object, or can be as complex as `load-depth-photo-from-network-and-save.js` checking DEP module supports XDMUtils interface (EXAMPLE 3 in DEP spec);
- run the tests via `xwalk.exe` with a `manifest.json` for extensions and http://localhost:8000/index.html which is enabled by a customized Python SimpleHTTPServer;
- send out the test results (JSON data) via `XMLHttpRequest()` to the server for printing out to standard output, so that CI system can catch these results to indicate a pass or fail;
- implement the customized Python SimpleHTTPServer with POST support (`rsserver.py`); it will deal with the test results, and shutdown itself after send out a 'no-cache' head and a response;
- include all the above operations into a Python script, `test.py`, so that we can simply enable the tests in CI system.

BUG=https://crosswalk-project.org/jira/browse/XWALK-5928

### CODE REVIEW

File: test/resources/testharnessreport.js [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+        ? document.body                 // <body> is required in XHTML documents
+        : document.documentElement;     // fallback for optional <body> in HTML5, SVG, etc.
+
+    parent.appendChild(results_element);
+
+    // Dump test results to server side and close the test.
+    var request = new XMLHttpRequest();
+    request.onreadystatechange = function () {
+        if (request.readyState == 4) {
+            window.close();
```
Comment (zqzhang) [inline test/resources/testharnessreport.js:411]: If you run the test 'http://localhost:8000/index.html' in Chrome, this line of code will return error `Scripts may not close windows that were not opened by script.`, and leave the test opened as-is.

If you run the test via `xwalk.exe`, the testing page may be closed; most times the page is opened there; thus cannot clean up the testing environment completely.

See https://groups.google.com/a/chromium.org/forum/#!topic/blink-dev/NZJQHRHrMpY

**Comment by zqzhang** [comment]
Complete this pull request now; please help review it. Thanks.

@leonhsl could you please help check whether it is OK in your local CI environment? Many thanks.

@DonnaWuDongxia could you please help review the README to see whether it is OK for you to create new tests and run them? Thanks. You can check it at

https://github.com/zqzhang/realsense-extensions-crosswalk/blob/XWALK-5928/test/README.md

File: test/resources/testharnessreport.js (hunk 1 above)
Comment (zqzhang) [inline test/resources/testharnessreport.js:411]: Since running the tests via `xwalk.exe` can close the test page, leave this as-is here.

**Comment by zqzhang** [comment]
@huningxin could you please help merge this pull request? Thanks.

Should there be any problem, I could address in following pull requests.

**Comment by huningxin** [comment]
Yes. Agree to merge. Thanks!

**Pull request merged**